### PR TITLE
[KernelGen][Nvidia] Add _batch_norm_with_update operator with Triton kernel

### PR DESCRIPTION
**Pull request by YufanPeter**
### Summary

Adds a implementation of `_batch_norm_with_update` (`torch.ops.aten._batch_norm_with_update`), the training-mode batch normalization operator that computes batch statistics, normalizes the input, and updates `running_mean` / `running_var` in place. Matching the aten schema, it returns the 4-tuple `(output, save_mean, save_invstd, reserve)`.

The forward kernel reuses the proven attorch-derived `batch_norm_forward_kernel` (the same kernel backing `flag_gems.batch_norm` and `_native_batch_norm_legit_functional`): a 1D grid over channels, each program running a serial Welford reduction over its `(batch × spatial)` slice, followed by a normalization pass in the same launch. Implementation details:

- Returns the 4-tuple `(output, save_mean, save_invstd, reserve)` matching the aten signature.
- `save_invstd` returns `inv_std = 1/sqrt(var + eps)`, following PyTorch's convention.
- `reserve` is an empty `uint8` tensor, consistent with the Triton fallback path (neither cuDNN nor MIOpen).
- `running_mean` and `running_var` are updated in place using the specified `momentum`.

### Testing

- Parametrized over 3D / 4D input shapes, affine and non-affine modes, and dtypes `float16 / float32 / bfloat16`
- Coverage: normalized output correctness, `save_mean` / `save_invstd` values, and in-place running-statistics updates
- **All 30 accuracy cases pass**

### Performance

Test command: `pytest benchmark/test__batch_norm_with_update.py --level core` (NVIDIA H20)

| dtype | shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|-------|--------------------|-------------------|---------|
| float16 | [4, 16, 64, 4] | 0.012672 | 0.006400 | 1.980 |
| float16 | [16, 16, 8, 48] | 0.019840 | 0.007776 | 2.551 |
| float16 | [16, 16, 8, 88] | 0.025088 | 0.010240 | 2.450 |
| float16 | [16, 16, 128] | 0.013984 | 0.006688 | 2.091 |
| float16 | [20, 6, 65536] | 1.070976 | 0.333760 | 3.209 |
| float32 | [4, 16, 64, 4] | 0.007424 | 0.006240 | 1.190 |
| float32 | [16, 16, 8, 48] | 0.009280 | 0.009024 | 1.028 |
| float32 | [16, 16, 8, 88] | 0.011776 | 0.011232 | 1.048 |
| float32 | [16, 16, 128] | 0.007808 | 0.006784 | 1.151 |
| float32 | [20, 6, 65536] | 0.590752 | 0.606784 | 0.974 |
| bfloat16 | [4, 16, 64, 4] | 0.013472 | 0.006400 | 2.105 |
| bfloat16 | [16, 16, 8, 48] | 0.018880 | 0.007840 | 2.408 |
| bfloat16 | [16, 16, 8, 88] | 0.025184 | 0.010304 | 2.444 |
| bfloat16 | [16, 16, 128] | 0.012736 | 0.006656 | 1.913 |
| bfloat16 | [20, 6, 65536] | 1.070112 | 0.844864 | 3.157 |

#### Arithmetic Mean Speedup

| dtype | Mean Speedup |
|-------|--------------|
| float16 | 2.456 |
| float32 | 1.108 |
| bfloat16 | 2.405 |
| **Overall** | **1.990x** |

### Multi-backend Testing

| Backend | Accuracy Test | Speedup (mean) | Notes |
|---|---|---|---|
| Nvidia (H20-3e) | PASS (30 cases) | 1.990 (overall) | Primary (fp16 / fp32 / bf16); channel-parallel kernel, best on large tensors |

### Files Changed

- `src/flag_gems/ops/_batch_norm_with_update.py`: Triton implementation (attorch-derived batch_norm forward kernel)
- `tests/test__batch_norm_with_update.py`: Accuracy test
- `benchmark/test__batch_norm_with_update.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to config
- `conf/operators.yaml`: Add operator entry (kind: NeuralNetwork, stage: alpha 5.4)

### CODE REVIEW

File: benchmark/test__batch_norm_with_update.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    weight = torch.randn((C,), dtype=dtype, device=device)
+    bias = torch.randn((C,), dtype=dtype, device=device)
+    running_mean = torch.randn((C,), dtype=dtype, device=device)
+    running_var = torch.abs(torch.randn((C,), dtype=dtype, device=device)) + 0.1
+    momentum = 0.1
+    eps = 1e-5
+    yield inp, weight, bias, running_mean, running_var, momentum, eps
+
+
+@pytest.mark._batch_norm_with_update
```
Comment (Caeruleann) [inline benchmark/test__batch_norm_with_update.py]:
```suggestion
@pytest.mark.batch_norm_with_update
```

File: conf/operators.yaml [hunk 2]
```diff
@@ -13,6 +13,17 @@
 # limitations under the License.
 
 ops:
+  - id: _batch_norm_with_update
```
Comment (Caeruleann) [inline conf/operators.yaml]:
```suggestion
  - id: batch_norm_with_update
```

File: tests/test__batch_norm_with_update.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+setattr(
+    pytest.mark,
+    "_batch_norm_with_update",
+    MarkDecorator(
+        Mark("_batch_norm_with_update", (), {}, _ispytest=True), _ispytest=True
+    ),
+)
+
+
+@pytest.mark._batch_norm_with_update
```
Comment (Caeruleann) [inline tests/test__batch_norm_with_update.py]:
```suggestion
@pytest.mark.batch_norm_with_update
```

File: conf/operators.yaml [hunk 4]
```diff
@@ -13,6 +13,10 @@
 # limitations under the License.
 
 ops:
+  - id: batch_norm_with_update
+    description: Applies training-mode Batch Normalization using batch statistics and updates running_mean/running_var in place.
+    for:
+      - _batch_norm_with_update
```
Comment (Caeruleann) [inline conf/operators.yaml:19]:
```suggestion
      - _batch_norm_with_update
    labels:
      - KernelGen
    stages:
      - alpha: '5.4'
```

File: tests/test__batch_norm_with_update.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    eps = 1e-5
+
+    ref_inp = utils.to_reference(inp, True)
+    ref_weight = utils.to_reference(weight, True)
+    ref_bias = utils.to_reference(bias, True)
+    ref_running_mean = utils.to_reference(running_mean.clone(), True)
+    ref_running_var = utils.to_reference(running_var.clone(), True)
+
+    # Reference: training-mode batch norm updates running stats in place.
+    ref_out = torch.nn.functional.batch_norm(
```
Comment (Caeruleann) [inline tests/test__batch_norm_with_update.py]:
```suggestion
    ref_out = torch.ops.aten._batch_norm_with_update((
```

keep same with gems mode.
